lib/tasks/get-space-data.js

```diff
@@ -107,6 +107,29 @@ export default function getFullSourceSpace ({
       }),
       skip: () => skipContent
     },
+    {
+      title: 'Fetching tags data',
```

Comment:
Maybe it's okay to try to fetch tags without confirming if the space has the feature enabled or not and the management sdk or the API should be figuring this out?

Fetching entries/assets before Tags which they rely on might get us into the same situation as fetching them without content types first.

If `getTags` returns an error or even an empty array because the feature isn't enabled, then we don't need entries or assets to determine this.
With this, we can change the order i.e. have fetching of tags come before fetching entries and assets because of the dependency content has on tags.

Reply:
> Fetching entries/assets before Tags which they rely on might get us into the same situation as fetching them without content types first.

I dont think this is a problem since we don't resolve links in the export. Though the dependency is annoying-- I agree that you shouldn't have to skip tags if you are skipping entries/assets. 

The one thing I want to preserve is the difference between omitting the tags array (if tags aren't enabled for you or you're using a cda token) and returning an empty array (some other error happened). This is consistent with other flags. Ex: `--skipContentModel` will omit the contentTypes array rather than return an empty one. If you don't have it enabled I think it's better to leave it out completely. Or is this inconsistent behavior just more confusing?


My proposal:
I think I'll `catch()` the error, if it's a 404 (not enabled) then I'll omit tags. Else I'll just return an empty array.

Reply:
and I'll skip the task with cdatoken (which omits the array).

Reply:
> I dont think this is a problem since we don't resolve links in the export

Is it possible for the exported file to contain the entries and assets but not tags because there was some interruption (network, etc) and the fetch tags task didn't get executed after entries/assets were fetched?

Regardless, I think with your proposal and skipping based on cdatoken, we can make the fetch tags task happen before the fetch entries and assets task, right?

Reply:
yeah good shout, I'll make tags export happen before entries and assets.

and my other proposal didn't work out as planned-- the tags array gets exported if the task is run. So it's actually very simple now:

* tag export is skipped based on the presence of cdaClient (which is only created in the case the cdaToken is passed)
* all other times it runs. if an error occurs (ex: 404 for not enabled, or something else), then the resulting export just has an empty tags array